hdinsight/resource-manager/v2018_06_01_preview/src/test/java/com/microsoft/azure/management/hdinsight/v2018_06_01_preview/utilities/HDInsightTestResourceManager.java

```diff
@@ -0,0 +1,296 @@
+     */
+    private final StorageManager storageManager;
+
+    /**
+     * Entry point to Azure Managed Service Identity (MSI) resource management.
+     */
+    private final MSIManager msiManager;
+
+    /**
+     * Entry point to Azure KeyVault resource management.
+     */
+    private final KeyVaultManager keyVaultManager;
+
+    /**
+     * Key vault client.
+     */
+    private final KeyVaultClient keyVaultClient;
+
+    private final String tenantId;
+
+    public HDInsightTestResourceManager(RestClient restClient, String subscription, String tenant) {
+        resourceManager = ResourceManager.authenticate(restClient).withSubscription(subscription);
+        storageManager = StorageManager.authenticate(restClient, subscription);
+        msiManager = MSIManager.authenticate(restClient, subscription);
+        keyVaultManager = KeyVaultManager.authenticate(restClient, tenant, subscription);
+        keyVaultClient = new KeyVaultClient(createKeyVaultRestClient(restClient));
+        tenantId = tenant;
+    }
+
+    /**
+     * Create REST client for Key vault client
+     */
+    private RestClient createKeyVaultRestClient(RestClient restClient) {
+        if (isRecordMode()) {
+            return restClient.newBuilder().withBaseUrl("https://{vaultBaseUrl}").build();
+        } else {
+            return restClient;
+        }
+    }
+
+    /**
+     * Create a resource group
+     */
+    public ResourceGroup createResourceGroup() {
+        return createResourceGroup(RESOURCE_GROUP_NAME_PREFIX, REGION);
+    }
+
+    public ResourceGroup createResourceGroup(String namePrefix, String region) {
+        String resourceGroupName = generateRandomResourceName(namePrefix, AZURE_RESOURCE_NAME_MAX_LEN);
+
+        // Here we assume that the resource group name is available.
+        // Considering that deletion is a dangerous operation, we just report error back to the tester
+        assertThat(resourceManager.resourceGroups().contain(resourceGroupName)).isFalse();
+
+        return resourceManager.resourceGroups().define(resourceGroupName)
+            .withRegion(region)
+            .create();
+    }
+
+    /**
+     * delete a resource group
+     */
+    public void deleteResourceGroup(String resourceGroupName) {
+        resourceManager.resourceGroups().beginDeleteByName(resourceGroupName);
+    }
+
+    /**
+     * Create a storage account.
+     */
+    public com.microsoft.azure.management.storage.StorageAccount createStorageAccount(String resourceGroupName) {
+        return createStorageAccount(resourceGroupName, STORAGE_ACCOUNT_NAME_PREFIX);
+    }
+
+    public com.microsoft.azure.management.storage.StorageAccount createStorageAccount(
+        String resourceGroupName,
+        String namePrefix) {
+        return createStorageAccount(resourceGroupName, namePrefix, STORAGE);
+    }
+
+    public com.microsoft.azure.management.storage.StorageAccount createStorageAccount(
+        String resourceGroupName,

```

Comment:
The style looks weird. 😄

Reply:
Hmmm. I will join parameter list into one line for a few methods with short parameter list. However, for methods with long parameter list, I think this style has better readability.